src/ComponentRegister.test.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+
+import {
+  ComponentRegister,
+  LinkComponent,
+  LinkComponentProps
+} from "./ComponentRegister";
+
+const TestLinkComponent: LinkComponent = ({
+  href,
+  children
+}: LinkComponentProps) => <div data-href={href}>{children}</div>;
+
+describe(".init()", () => {
+  it("registers all of the components provided", () => {
+    ComponentRegister.init({
+      components: {
+        Link: TestLinkComponent
+      }
+    });
+
+    expect(ComponentRegister.Link).toStrictEqual(TestLinkComponent);
+  });
+  it("does not alter properties if none are provided", () => {
+    ComponentRegister.init({});
```

Comment:
```suggestion
    ComponentRegister.init();
```

Let's make sure it's possible to call without any arguments. This might be better as a separate test. I leave that up to you.